File: src/main/scala/tao/TAOInstance.scala
package tao

class TAOInstance(
  val nAssistants: Int,
  val maxHours: Array[Int],
  val nCourses: Int,
  val hours: Array[Int],
  val courseAssistants: Array[Array[Int]],
  val required: Array[(Int, Int)],
  val forbidden: Array[(Int, Int)],
  val oldSolution: Array[Int]
)

File: src/main/java/tao/TAOInstance.java
package tao;

import tao.util.Pair;

public class TAOInstance {
	
	public final int nAssistants;
	public final int nCourses;
	public final int[] maxHours;
	public final int[] hours;
	public final int[][] courseAssistants;
	public final Pair[] required;
	public final Pair[] forbidden;
	public final int[] oldAssignments;
	
	public TAOInstance(int[] maxHours, int[] hours, int[][] courseAssistants, Pair[] required, Pair[] forbidden, int[] oldAssignments) {
		this.nAssistants = maxHours.length;
		this.nCourses = courseAssistants.length;
		this.maxHours = maxHours;
		this.hours = hours;
		this.courseAssistants = courseAssistants;
		this.required = required;
		this.forbidden = forbidden;
		this.oldAssignments = oldAssignments;
	}
}

